pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -1289,6 +1237,75 @@ func (c *Cloud) ensureLoadBalancer(namespacedName types.NamespacedName, loadBala
 	return loadBalancer, nil
 }
 
+// syncElbListeners computes a plan to reconcile the desired vs actual state of the listeners on an ELB
+// NOTE: there exists an O(nlgn) implementation for this function. However, as the default limit of
+//       listeners per elb is 100, this implementation is reduced from O(m*n) => O(n).
+func syncElbListeners(loadBalancerName string, listeners []*elb.Listener, listenerDescriptions []*elb.ListenerDescription) (*elb.CreateLoadBalancerListenersInput, *elb.DeleteLoadBalancerListenersInput) {
+	foundSet := make(map[int]bool)
+	deleteRequest := &elb.DeleteLoadBalancerListenersInput{
+		LoadBalancerName: aws.String(loadBalancerName),
+	}
+	createRequest := &elb.CreateLoadBalancerListenersInput{
+		LoadBalancerName: aws.String(loadBalancerName),
+	}
+
+	for _, listenerDescription := range listenerDescriptions {
+		actual := listenerDescription.Listener
+		if actual == nil {
+			klog.Warning("Ignoring empty listener in AWS loadbalancer: ", loadBalancerName)
+			continue
+		}
+
+		for i, expected := range listeners {
+			if elbListenersAreEqual(actual, expected) {
+				// The current listener on the actual
+				// elb is in the set of desired listeners.
+				foundSet[i] = true
+				break
+			}
+			if i == len(listeners)-1 {
```

Comment:
A bit confused by this check, can you expand?

Reply:
Here is the old logic:
```
if found != -1 {
	foundSet[found] = true
} else {
	removals = append(removals, actual.LoadBalancerPort)
}
```
If we haven't broken out of the loop by now and are on the last element in the list, we know that the listener isn't part of the desired state. Hence, we can mark it for deletion.

This implementation removes the necessity for the previously needed dirty bit, `found`.

Reply:
I see, it makes the code much harder to understand IMO and I don't think the trade-off is worth it here